src/whispercpp/context.cc

```diff
@@ -316,7 +317,51 @@ float Context::full_get_token_prob(int segment, int token) {
   return whisper_full_get_token_p(ctx, segment, token);
 }
 
+    // typedef struct whisper_token_data {
+    //     whisper_token id;  // token id
+    //     whisper_token tid; // forced timestamp token id
+
+    //     float p;           // probability of the token
+    //     float plog;        // log probability of the token
+    //     float pt;          // probability of the timestamp token
+    //     float ptsum;       // sum of probabilities of all timestamp tokens
+
+    //     // token-level timestamp data
+    //     // do not use if you haven't computed token-level timestamps
+    //     int64_t t0;        // start time of the token
+    //     int64_t t1;        //   end time of the token
+
+    //     float vlen;        // voice length of the token
+    // } whisper_token_data;
+
+
 void ExportContextApi(py::module &m) {
+  py::class_<whisper_token_data>(m, "WhisperTokenData", "Data for the token")
+    .def_readonly("id", &whisper_token_data::id)
+    .def_readonly("tid", &whisper_token_data::tid)
+    .def_readonly("p", &whisper_token_data::p)
+    .def_readonly("plog", &whisper_token_data::plog)
+    .def_readonly("pt", &whisper_token_data::pt)
+    .def_readonly("ptsum", &whisper_token_data::ptsum)
+    .def_readonly("t0", &whisper_token_data::t0)
+    .def_readonly("t1", &whisper_token_data::t1)
+    .def_readonly("vlen", &whisper_token_data::vlen)
+    .def("__repr__", [](const whisper_token_data &t) {
+      std::stringstream s;
+      s << "("
+        << "id=" << t.id << ", "
+        << "tid=" << t.tid << ", "
+        << "p=" << t.p << ", "
+        << "plog=" << t.plog << ", "
+        << "pt=" << t.pt << ", "
+        << "ptsum=" << t.ptsum << ", "
+        << "t0=" << t.t0 << ", "
+        << "t1=" << t.t1 << ", "
+        << "vlen=" << t.vlen << ")";
+
+      return s.str();
+    });
```

Comment:
I will probably write a template or something to make this process more streamline. Would prefer not have to write __repr__ for all of the exported class :smile:

I think for this case it is ok since `whisper_token_data` is relatively small.

Reply:
Sure a template for that sounds nice. 
I can also drop the repr, was just handy for testing :sweat_smile:

Reply:
no worries. it is fine. the code is already there no need to do anything extra.